agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFilePurger.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.agent.internal.localstorage;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import com.microsoft.applicationinsights.agent.internal.common.ThreadPoolUtils;
+import java.io.File;
+import java.util.Collection;
+import java.util.Date;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Purge files that are older than 48 hours in both 'telemetry' and 'statsbeat' folders. Purge is
+ * run every 24 hours.
+ */
+public class LocalFilePurger implements Runnable {
+
+  private static final Logger logger = LoggerFactory.getLogger(LocalFilePurger.class);
+
+  private final long expiredInterval;
+  private final File folder;
+
+  public static void startPurging() {
+    startPurging(null, null, null);
+  }
+
+  // this is used by tests to configure purge interval, expired interval and the test folder which
+  // files are to be purged.
+  static void startPurging(
+      @Nullable Long purgeIntervalSeconds, @Nullable Long expiredIntervalSeconds, File folder) {
+    long interval =
+        purgeIntervalSeconds == null ? TimeUnit.DAYS.toSeconds(1) : purgeIntervalSeconds;
+    long expiredInterval =
+        expiredIntervalSeconds == null ? TimeUnit.DAYS.toSeconds(2) : expiredIntervalSeconds;
```

Comment:
I think more conventional to pass defaults directly instead of passing `null`

Reply:
this overloading is merely for testing and it's package protected.

Reply:
sent commit to your branch to show how this change makes things simpler, reducing conditionals and code paths

Reply:
yup, it doesn't require updating the tests.